bridge/svix-bridge/src/config/tests.rs

```diff
@@ -331,3 +334,174 @@ fn test_senders_example() {
     assert!(!conf.senders.is_empty());
     assert!(conf.receivers.is_empty());
 }
+
+#[test]
+fn test_multiline_template_detection() {
+    let src = r#"
+senders:
+  - name: "rabbitmq-example"
+    input:
+      type: "rabbitmq"
+      # Required
+      uri: "amqp://guest:guest@localhost:5672/%2f"
+      # Required
+      queue_name: "${QUEUE_NAME}"
+      # Optional (default: unset, managed by rabbitmq)
+      consumer_tag: "my-consumer-001"
+      # Optional: default true
+      requeue_on_nack: true
+"#;
+    assert!(super::is_templated(src));
+}
+
+#[test]
+fn test_variable_substitution_missing_vars() {
+    let src = r#"
+    opentelemetry: 
+        address: "${OTEL_ADDR}"
+    "#;
+    let vars = HashMap::new();
+    let cfg = Config::from_src(src, Some(&vars)).unwrap();
+    let otel = cfg.opentelemetry.unwrap();
+    // when lookups in the vars map fail, the original token text is preserved.
+    assert_eq!(&otel.address, "${OTEL_ADDR}");
+}
+
+#[test]
+fn test_variable_substitution_available_vars() {
+    let src = r#"
+    opentelemetry:
+        address: "${OTEL_ADDR}"
+        sample_ratio: ${OTEL_SAMPLE_RATIO}
+    "#;
+    let mut vars = HashMap::new();
+    vars.insert(
+        String::from("OTEL_ADDR"),
+        String::from("http://127.0.0.1:8080"),
+    );
+    vars.insert(String::from("OTEL_SAMPLE_RATIO"), String::from("0.25"));
+    let cfg = Config::from_src(src, Some(&vars)).unwrap();
+    // when lookups succeed, the token should be replaced.
+    let otel = cfg.opentelemetry.unwrap();
+    assert_eq!(&otel.address, "http://127.0.0.1:8080");
+    assert_eq!(otel.sample_ratio, Some(0.25));
+}
+
+#[test]
+fn test_variable_substitution_requires_braces() {
+    let src = r#"
+    opentelemetry:
+        # Neglecting to use ${} notation means the port number will not be substituted. 
+        address: "${OTEL_SCHEME}://${OTEL_HOST}:$OTEL_PORT"
+    "#;
+    let mut vars = HashMap::new();
+    vars.insert(String::from("OTEL_SCHEME"), String::from("https"));
+    vars.insert(String::from("OTEL_HOST"), String::from("127.0.0.1"));
+    vars.insert(String::from("OTEL_PORT"), String::from("9999"));
+    let cfg = Config::from_src(src, Some(&vars)).unwrap();
+    // when lookups succeed, the token should be replaced.
+    let otel = cfg.opentelemetry.unwrap();
+    // Not the user-intended outcome, but it simplifies the parsing requirements.
+    assert_eq!(&otel.address, "https://127.0.0.1:$OTEL_PORT");
+}
+
+#[test]
+fn test_variable_substitution_missing_numeric_var_is_err() {
+    // FIXME: discuss in code review
```

Comment:
The nutshell here is: when a var is specified in the config but is not set in the env, the current behavior is to leave the original var as-is.

Example:
```yaml
queue_name: "${QUEUE_NAME}"
```

For string fields, this config will parse just fine because the literal string `"${QUEUE_NAME}"` works as a string, but will error if we try to connect to a bad queue name.

If we wanted to error immediately when a var is missing (could be better), I'd probably ditch the dep on `envsubst` and rewrite it in here with that behavior.

Reply:
Nah, I think this is great, as it's the least likely to clash with other things.

Does this lib also support escaping?

Reply:
Sadly, nope. If there's not really anything to purposefully prevent it from trying to do a replacement.

Some other systems use `$${...}` to retain the literal instead of doing a substitution. Not supported here 😓

Reply:
I think it's fine anyway, as clashes would be unlikely?